scheduler-server/src/main/java/gov/nasa/jpl/aerie/scheduler/server/remotes/postgres/SetRequestStateAction.java

```diff
@@ -28,12 +32,18 @@ public void apply(
       final long specificationId,
       final long specificationRevision,
       final RequestRecord.Status status,
-      final ScheduleFailure failureReason
+      final ScheduleFailure failureReason,
+      final Optional<DatasetId> datasetId
   ) throws SQLException {
     this.statement.setString(1, status.label);
     PreparedStatements.setFailureReason(this.statement, 2, failureReason);
-    this.statement.setLong(3, specificationId);
-    this.statement.setLong(4, specificationRevision);
+    if(datasetId.isPresent()) {
+      statement.setLong(3, datasetId.get().id());
+    } else{
+      statement.setNull(3, Types.BIGINT);
```

Comment:
Should be:

```suggestion
      statement.setNull(3, Types.INTEGER);
```

In order to match `scheduling_request.dataset`'s type.